src/NSwag.SwaggerGeneration.AspNetCore/Processors/OperationParameterProcessor.cs

```diff
@@ -87,10 +87,7 @@ public OperationParameterProcessor(AspNetCoreToSwaggerGeneratorSettings settings
                 {
                     var operationParameter = await CreatePrimitiveParameterAsync(context, extendedApiParameter).ConfigureAwait(false);
                     operationParameter.Kind = SwaggerParameterKind.Path;
-                    operationParameter.IsRequired = true; // Path is always required => property not needed
-
-                    if (_settings.SchemaType == SchemaType.Swagger2)
-                        operationParameter.IsNullableRaw = false;
+                    operationParameter.IsRequired = !apiParameter.RouteInfo.IsOptional;
```

Comment:
@rynowak

Reply:
In Swagger, path parameters must be required... not sure if this will produce invalid specs...

Reply:
I thought that's what the existing comment indicated. However, Mvc does allow optional route tokens at the end of the route e.g. `[controller]/[action]/{id?}` where id is optional. Is there a better way to represent it?

Reply:
Note that I'm aware of. MVC has lots of constructs that don't fit swagger. I think the answer is that the swagger should say it's required. If the implementation is a superset of the documentation, that's not ideal, but it doesn't break reality.